### PR TITLE
Add unit tests for `ITransport`

### PR DESCRIPTION
**Pull request by limebell**
Adds unit tests for `ITransport` for further extensions, and bumps C# version of `Libplanet.Tests` project to 8.0.

### CODE REVIEW

**Review by riemannulus — DISMISSED** [review]

File: Libplanet/Net/Transports/TransportException.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        {
+        }
+
+        protected TransportException(
+            SerializationInfo info,
+            StreamingContext context
+        )
+            : base(info, context)
+        {
+        }
```
Comment (moreal) [inline Libplanet/Net/Transports/TransportException.cs:30]: Could you add unit test for serialize, deserialize, like [Libplanet.Tests/Blocks/InvalidBlocckTxHashExceptionTest.cs](https://github.com/planetarium/libplanet/blob/2301630e615ab2ac301a4babe3a3012ffc46fa4a/Libplanet.Tests/Blocks/InvalidBlocckTxHashExceptionTest.cs) does?

**Review by moreal — APPROVED** [review]

File: Libplanet/Net/Transports/TransportException.cs (hunk 1 above)
Comment (limebell) [reply Libplanet/Net/Transports/TransportException.cs:30]: Great, I'll amend it.

**Review by moreal — DISMISSED** [review]

**Review by dahlia — COMMENTED** [review]
The commits are based on the outdated *main* branch.  Need to rebase.

**Review by dahlia — DISMISSED** [review]

File: Libplanet/Net/Transports/NetMQTransport.cs [hunk 2]
```diff
@@ -518,6 +524,7 @@ public void ReplyMessage(Message message)
 
         private void ReceiveMessage(object sender, NetMQSocketEventArgs e)
         {
+            _logger.Verbose("ReceiveMessage started. {@E}", e);
```
Comment (moreal) [inline Libplanet/Net/Transports/NetMQTransport.cs:527]: How about using `nameof` keyword?

```suggestion
            _logger.Verbose($"{nameof(ReceiveMessage)} started. {{@E}}", e);
```

File: Libplanet/Net/Transports/NetMQTransport.cs [hunk 3]
```diff
@@ -586,6 +593,8 @@ private void ReceiveMessage(object sender, NetMQSocketEventArgs e)
                     );
                 }
             }
+
+            _logger.Verbose("ReceiveMessage ended. {@E}", e);
```
Comment (moreal) [inline Libplanet/Net/Transports/NetMQTransport.cs:597]: How about using `nameof` keyword?

```suggestion
            _logger.Verbose($"{nameof(ReceiveMessage)} ended. {{@E}}", e);
```

**Review by moreal — APPROVED** [review]

File: Libplanet/Net/Transports/NetMQTransport.cs (hunk 2 above)
Comment (limebell) [reply Libplanet/Net/Transports/NetMQTransport.cs:527]: Thanks for the comment, but sadly this line was used in local testing and added to vcs by my mistake 😢 . I'll remove this.

File: Libplanet/Net/Transports/NetMQTransport.cs (hunk 3 above)
Comment (limebell) [reply Libplanet/Net/Transports/NetMQTransport.cs:597]: Same with discussion [above](https://github.com/planetarium/libplanet/pull/1242#discussion_r621902603).

**Review by dahlia — DISMISSED** [review]

**Review by dahlia — DISMISSED** [review]

**Review by longfin — DISMISSED** [review]

**Pull request merged**